tooling/templatize/pkg/pipeline/shell.go

```diff
@@ -13,26 +13,30 @@ import (
 )
 
 func (s *Step) createCommand(ctx context.Context, dryRun bool, envVars map[string]string) (*exec.Cmd, bool) {
-	var cmd *exec.Cmd
+	var scriptCommand string = s.Command
 	if dryRun {
-		if s.DryRun.Command == nil && s.DryRun.EnvVars == nil {
+		if s.DryRun.Command == "" && s.DryRun.EnvVars == nil {
 			return nil, true
 		}
+		if s.DryRun.Command != "" {
+			scriptCommand = s.DryRun.Command
+		}
 		for _, e := range s.DryRun.EnvVars {
 			envVars[e.Name] = e.Value
 		}
-		if s.DryRun.Command != nil {
-			cmd = exec.CommandContext(ctx, s.DryRun.Command[0], s.DryRun.Command[1:]...)
-		}
-	}
-	if cmd == nil {
-		// if dry-run is not enabled, use the actual command or also if no dry-run command is defined
-		cmd = exec.CommandContext(ctx, s.Command[0], s.Command[1:]...)
 	}
+	cmd := exec.CommandContext(ctx, "/bin/bash", "-c", buildBashScript(scriptCommand))
 	cmd.Env = append(cmd.Env, utils.MapToEnvVarArray(envVars)...)
 	return cmd, false
 }
 
+func buildBashScript(command string) string {
+	return "set -o errexit\n" +
```

Comment:
Did you consider strings.Builder? https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/47798475#47798475

Reply:
i've opted for simply fmt.Sprintf